albatross/core/prediction.h

```diff
@@ -150,9 +143,28 @@ class Prediction {
   Eigen::VectorXd
   joint() const = delete; // No valid predict joint method found.
 
+  template <typename PredictType>
+  PredictType get(PredictTypeIdentity<PredictType> =
```

Comment:
mostly it just makes it so you don't have redundency when actually call this method.  Ie,

    `prediction.get<MarginalDistribution>();`

instead of

    `prediction.get<MarginalDistribution>(PredictTypeIdentity<MarginalDistribution>());`

Reply:
I meant why can't we get rid of the argument all together? So you'd still be able to do
```prediction.get<MarginalDistribution>();```

Reply:
Ah, yeah I see.  The extra parameter is necessary since the return types are different.